plugin/trino-open-policy-agent/src/main/java/io/trino/plugin/openpolicyagent/OpaAuthorizer.java

```diff
@@ -0,0 +1,827 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openpolicyagent;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.google.inject.Inject;
+import io.trino.spi.connector.CatalogSchemaName;
+import io.trino.spi.connector.CatalogSchemaRoutineName;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.function.FunctionKind;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.Privilege;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.spi.security.SystemSecurityContext;
+import io.trino.spi.security.TrinoPrincipal;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class OpaAuthorizer
+        implements SystemAccessControl
+{
+    private final HttpClient httpClient;
+    private final ObjectMapper json;
+    private final URI opaPolicyUri;
+
+    @Inject
+    public OpaAuthorizer(OpaConfig config)
+    {
+        this(config, HttpClient.newHttpClient());
+    }
+
+    public OpaAuthorizer(OpaConfig config, HttpClient httpClient)
+    {
+        this.opaPolicyUri = config.getOpaUri();
+        this.json = new ObjectMapper();
+        // do not include null values
+        this.json.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        // deal with Optional<T> values
+        this.json.registerModule(new Jdk8Module());
```

Comment:
```suggestion
        this.json = new ObjectMapper()
                // OPA server can send other fields, such as `decision_id` when enabling decision logs
                // We could add all the fields we *currently* know, but it's more future-proof to ignore any unknown fields.
                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
                // Do not include null values
                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
                // Deal with Optional<T> values
                .registerModule(new Jdk8Module());
```

Reply:
The latest commit no longer uses the ObjectMapper and instead relies on the airlift JSON module (which is based on jackson too)